kernel/syscall.c

```diff
@@ -6624,37 +6654,80 @@ long myst_syscall_tgkill(int tgid, int tid, int sig)
     return ret;
 }
 
-long myst_syscall_kill(int pid, int sig)
+static long _myst_send_kill(myst_process_t* process, int signum)
 {
     long ret = 0;
-    myst_thread_t* thread = myst_thread_self();
-    myst_process_t* process = NULL;
+    siginfo_t* siginfo;
 
-    myst_spin_lock(&myst_process_list_lock);
+    if (!(siginfo = calloc(1, sizeof(siginfo_t))))
+        ERAISE(-ENOMEM);
 
-    // Find process thread from pid
-    process = myst_find_process_from_pid(pid, false);
+    siginfo->si_code = SI_USER;
+    siginfo->si_signo = signum;
+    siginfo->si_pid = process->pid;
+    siginfo->si_uid = process->main_process_thread->euid;
 
-    // Did we finally find it?
-    if (process == NULL)
-        ERAISE(-ESRCH);
+    ret = myst_signal_deliver(process->main_process_thread, signum, siginfo);
```

Comment:
If ``myst_signal_deliver()`` fails, maybe ``siginfo`` is leaked.

Reply:
This is an ugliness of myst_signal_deliver. It always deletes the buffer even if failed.